File: tests/machine/x/dart/membership.dart
bool _in(dynamic item, dynamic col) {
  if (col is Map) return col.containsKey(item);
  if (col is Iterable || col is String) return col.contains(item);
  return false;
}

void main() {
  var nums = [1, 2, 3];
  print(_in(2, nums));
  print(_in(4, nums));
}

File: tests/machine/x/java/membership.java
import java.util.*;
public class Membership {
	static List<Integer> nums = new ArrayList<>(Arrays.asList(1, 2, 3));
	static boolean inOp(Object item, Object collection) {
		if (collection instanceof Map<?,?> m) return m.containsKey(item);
		if (collection instanceof Collection<?> c) return c.contains(item);
		if (collection instanceof String s) return s.contains(String.valueOf(item));
		return false;
	}
	public static void main(String[] args) {
	System.out.println(inOp(2, nums));
	System.out.println(inOp(4, nums));
	}
}
